plugins/kafka-backend/src/service/router.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { Config } from '@backstage/config';
+import { KafkaApi, KafkaJsApiImpl } from './KafkaApi';
+import _ from 'lodash';
+
+export interface RouterOptions {
+  logger: Logger;
+  config: Config;
+}
+
+export const makeRouter = (
+  logger: Logger,
+  kafkaApi: KafkaApi,
+): express.Router => {
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/consumer/:consumerId/offsets', async (req, res) => {
+    const consumerId = req.params.consumerId;
+
+    logger.debug(`Fetch consumer group ${consumerId} offsets`);
+
+    const groupOffsets = await kafkaApi.fetchGroupOffsets(consumerId);
+
+    const groupWithTopicOffsets = await Promise.all(
```

Comment:
Hm, so I tried making the API class just a thin wrapper on top of Kafka.js (so it'd be easier to switch it to a different library if someone wants to in the future), and then keeping more complex logic out of it. Maybe it's worth adding another layer here to do this calculations but I think since it's pretty slim for now it can be kept here, WDYT?

Reply:
Yeah, it's probably fine for now. But maybe a future consideration.